packages/tezos-protocol-environment/tezos-protocol-environment.15.0/opam

```diff
@@ -0,0 +1,57 @@
+opam-version: "2.0"
+maintainer: "contact@tezos.com"
+authors: ["Tezos devteam"]
+homepage: "https://www.tezos.com/"
+bug-reports: "https://gitlab.com/tezos/tezos/issues"
+dev-repo: "git+https://gitlab.com/tezos/tezos.git"
+license: "MIT"
+depends: [
+  "dune" { >= "3.0" }
+  "ocaml" { >= "4.14.0" & < "4.15" }
```

Comment:
Is there any reason why this is still not compatible with OCaml 5.0?

Reply:
The octez suite as a whole is not compatibly because it depends on `ringo` which we are still in the process of updating. (The release still depends on ringo<1.0.0 which is incompatible with ocaml5 because it iterates over an ephemeron table which is not domain-safe and thus removed in ocaml5.)

Reply:
Perhaps more fundamentally, we want to be very careful not to change the semantics of the economic protocol of Tezos, since that would introduce a fork in the blockchain. To this end, the protocol environment (i.e. this package) specifies exactly which major version of OCaml to use. Before bumping this major version we run tests to check that the chain still behaves the same.